testsuite/MDAnalysisTests/analysis/test_hydrogenbonds_analysis.py

```diff
@@ -204,6 +204,198 @@ def test_no_bond_info_exception(self, universe):
+                        [2.250, 0.968, 0]       # H4
+                        ])
+
+        # Atomic coordinates with no hydrogen bonds
+        pos2 = np.array([[0, 0, 0],             # O1
+                         [-0.249, -0.968, 0],   # H1
+                         [1, 0, 0],             # H2
+                         [4.5, 0, 0],           # O2
+                         [5., 0, 0],            # H3
+                         [4.250, 0.968, 0]      # H4
+                         ])
+
+        coordinates = np.empty((3,  # number of frames
+                                u.atoms.n_atoms,
+                                3))
+        coordinates[0] = pos1
+        coordinates[1] = pos2
+        coordinates[2] = pos1
+        u.load_new(coordinates, order='fac')
+
+        return u
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def hydrogen_bonds(universe):
+        h = HydrogenBondAnalysis(
+            universe,
+            **TestHydrogenBondAnalysisNoRes.kwargs
+        )
+        h.run()
+        return h
+
+    def test_count_by_type(self, universe):
+
+        h = HydrogenBondAnalysis(
+            universe,
+            **TestHydrogenBondAnalysisNoRes.kwargs
+        )
+        h.run()
+        counts = h.count_by_type()
+        ref_count = 2
+        assert int(counts[0, 2]) == ref_count
+
+    def test_no_bond_info_exception(self, universe):
+
+        kwargs = {
+            'donors_sel': None,
+            'hydrogens_sel': None,
+            'acceptors_sel': None,
+            'd_h_cutoff': 1.2,
+            'd_a_cutoff': 3.0,
+            'd_h_a_angle_cutoff': 120.0
+        }
+
+        with pytest.raises(NoDataError, match="no bond information"):
+            h = HydrogenBondAnalysis(universe, **kwargs)
+            h._get_dh_pairs()
+
+    def test_no_bond_no_mass_donor_sel(self, universe):
+
+        kwargs = {
+            'donors_sel': "type O",
+            'hydrogens_sel': None,
+            'acceptors_sel': None,
+            'd_h_cutoff': 1.2,
+            'd_a_cutoff': 3.0,
+            'd_h_a_angle_cutoff': 120.0
+        }
+
+        with pytest.raises(NoDataError, match="not contain mass information"):
+            h = HydrogenBondAnalysis(universe, **kwargs)
+            h._get_dh_pairs()
+
+    def test_no_bond_donor_sel(self, universe):
+
+        kwargs = {
+            'donors_sel': "type O",
+            'hydrogens_sel': None,
+            'acceptors_sel': None,
+            'd_h_cutoff': 1.2,
+            'd_a_cutoff': 3.0,
+            'd_h_a_angle_cutoff': 120.0
+        }
+        u = universe.copy()
+        n_residues = 2
+        u.add_TopologyAttr('mass', [15.999, 1.008, 1.008] * n_residues)
+        u.add_TopologyAttr('charge', [-1.04, 0.52, 0.52] * n_residues)
+        h = HydrogenBondAnalysis(u, **kwargs)
+        donors = u.select_atoms(h.guess_donors())
+
+        assert len(donors) == 2
+
     def test_first_hbond(self, hydrogen_bonds):
```

Comment:
This used to be part of the `TestHydrogenBondAnalysisIdeal` is there a reason for moving it to `TestHydrogenBondAnalysisNoRes` isntead?

Reply:
I'm confused on why you said "used to be" since `TestHydrogenBondAnalysisIdeal` hasn't been touched. I did repeat it but you'll notice in the "add_TopologyAttr" I used types instead of resnames and names. This is also the case in the method selection criteria. Although the system in the two universes is identical, the format of the universe in `TestHydrogenBondAnalysisIdeal` represents something imported from CHARMM or AMBER etc. and the one I made in `TestHydrogenBondAnalysisNoRes` represents that system after importing from LAMMPS.

Reply:
I see, thanks for the explanation, I think it's the diff messing with how things look.